src/hooks/useSwapStats.ts

```diff
@@ -0,0 +1,85 @@
+import { BLOCK_TIME, ChainId } from "../constants"
+import { useCallback } from "react"
+import { useQuery } from "@tanstack/react-query"
+
+const swapStatsURI = "https://ipfs.saddle.exchange/swap-stats.json"
+
+type PoolStats = {
+  oneDayVolume: string
+  apy: string
+  tvl: string
+  utilization: string
+}
+type SwapStats = {
+  [chainId in ChainId]?: Partial<{
+    [swapAddress: string]: PoolStats
+  }>
+}
+
+type SwapStatsReponse = {
+  [chainId in ChainId]?: {
+    [swapAddress: string]: {
+      oneDayVolume: number
+      APY: number
+      TVL: number
+    }
+  }
+}
+export function useSwapStats() {
+  return useQuery(
+    ["swapStats"],
+    async () => {
+      const res = await fetch(`${swapStatsURI}`)
+      if (res.status >= 200 && res.status < 300) {
+        return res.json()
+      }
+      throw new Error("Unable to fetch swap stats from IPFS")
+    },
+    {
+      select: useCallback(
+        (data: SwapStatsReponse) => formatSwapStats(data),
+        [],
+      ),
+      retry: 3,
+      refetchInterval: BLOCK_TIME * 3,
+    },
```

Comment:
I think you also need to add arguments for either `staleTime` or `cacheTime`, since the call is made every time that the hook is called. Verify this yourself by navigating to one pool's deposit page, then to another pool's deposit page. You will see two requests for the same data.

Reply:
I think some of unexpected re-fetch is occurring from `refetchOnWindowFocus`. I disable this feature.